app/src/main/java/com/rockspin/androiddevtest/MainActivity.java

```diff
@@ -22,11 +37,85 @@ protected void onCreate(Bundle savedInstanceState) {
         mRecyclerView = (RecyclerView) findViewById(R.id.recycler_view);
         mRecyclerView.setLayoutManager(new LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false));
 
+        m_sortButton = (FloatingActionButton) findViewById(R.id.fab_change_sorting);
+        m_sortButton.setOnClickListener(this);
+
         mAdapter = new EVActivityAdapter(this);
         mRecyclerView.setAdapter(mAdapter);
 
-        final Response<List<CosmonautActivity>> response = MyApplication.getAPIService().getEVList().execute();
-        final List<CosmonautActivity> cosmonautActivityList = response.body();
-        mAdapter.setCosmonautActivityList(cosmonautActivityList);
+
+        m_brcUpdateData = new BroadcastDataUpdateReceiver();
+        registerReceiver( m_brcUpdateData, new IntentFilter( MyApplication.BROADCAST_UPDATE_DATA_HANDLER ) );
+
+        if( hasInternetConnection() )
+        {
+            fetchData();
+            return;
+        }
+
+        AlertDialog.Builder builder = new AlertDialog.Builder( this );
+        builder.setMessage( getText( R.string.no_net_connection ) );
+        builder.setPositiveButton(getText(R.string.retry), new DialogInterface.OnClickListener()
+        {
+            @Override
+            public void onClick(DialogInterface dialog, int which)
+            {
+                m_alertDialog.dismiss();
+                fetchData();
+            }
+        });
+
+        m_alertDialog = builder.create();
+        m_alertDialog.setCancelable( false );
+        m_alertDialog.show();
+    }
+
+    protected void onDestroy()
+    {
+        super.onDestroy();
+        try
+        {
+            if( m_brcUpdateData != null )
```

Comment:
I prefer to have the check. In future the initialization of the broadcastreceiver can be changed.